### PR TITLE
feature: add PauliString sums

### LINKED ISSUE #1256: Allow for Sums of PauliStrings
This issue is part of **unitaryHACK26**. You have to be [registered](https://unitaryhack.dev/register/) to complete this issue.
Learn more about the PR submission process [here](https://unitaryhack.dev/hacker-guide/) and about unitaryHACK rules [here](https://unitary.design/rules/)!

**A note about AI Slop:** while we are open to collaboration with LLMs for unitaryHACK, fully AI-generated PRs are not acceptable. It is at the maintainers' discretion whether or not LLM-generated PRs are the right fit for the issues, and those that appear fully AI-generated may be immediately rejected. Read unitaryHACK's full AI Policy [here](https://unitaryhack.dev/ai-guide/) 

---

**Describe the feature you'd like**

The PauliString class contains logic for creating and manipulating PauliStrings, though only exists in single instances. The proposed class is a container for PauliStrings, which allows for:

1. Addition and subtraction of Pauli Strings, including with weights
2. Multiplication by floats
3. Multiplication by PauliStrings
4. Single-item index-ability & inclusion checks
5. Conversion to & from Sum & list of tuple[float, str]
6. [extra] Inter and extra-class operations (i.e. commutes with another operator, self commuting, etc.) 
This may require changes to PauliString as well, depending on how coefficients are treated.

### PR DESCRIPTION
**Pull request by dorakingx**
Closes #1256.

## Summary
- Adds `PauliSum`, a weighted container for `PauliString` terms.
- Supports Pauli string addition/subtraction, scalar multiplication, distributed multiplication by Pauli strings or other Pauli sums, indexing, membership checks, and commuting checks.
- Adds conversion helpers to and from weighted term lists and Braket `Observable.Sum` objects built from Pauli observables.
- Re-exports `PauliSum` from `braket.quantum_information`.

## Validation
- `uv run pytest test/unit_tests/braket/quantum_information/test_pauli_string.py -q`
- `uvx ruff check src/braket/quantum_information/pauli_string.py src/braket/quantum_information/__init__.py`
- `uvx ruff format --check src/braket/quantum_information/pauli_string.py src/braket/quantum_information/__init__.py test/unit_tests/braket/quantum_information/test_pauli_string.py`

AI assistance disclosure: I used Codex as a coding assistant, then inspected the generated changes and ran the validation commands above.

### CODE REVIEW

**Comment by dorakingx** [comment]
I pushed an update to address the Codecov failures.

Changes:
- Added tests for PauliSum edge cases and error paths that Codecov reported as missing.
- Added support for `PauliString * PauliSum` so multiplication distributes consistently in both directions.

Local validation:

```bash
uv run python -m coverage erase
PYTEST_ADDOPTS="" uv run python -m coverage run -m pytest -o addopts="" test/unit_tests/braket/quantum_information/test_pauli_string.py -q
uv run python -m coverage report -m src/braket/quantum_information/pauli_string.py
```

This now reports `src/braket/quantum_information/pauli_string.py` at 100% coverage locally.

I also ran:

```bash
uvx tox -e linters_check
uvx tox -e unit-tests -- test/unit_tests/braket/quantum_information/test_pauli_string.py -q
```

The targeted tox unit-test run passed with 65 passed and 32 xfailed.

File: src/braket/quantum_information/pauli_string.py [hunk 1]
```diff
@@ -408,3 +437,230 @@ def _generate_eigenstate_circuit(self, signs: tuple[int, ...]) -> Circuit:
...
+
+    def commutes_with(self, other: str | PauliString | PauliSum) -> bool:
+        """Returns `True` if every term commutes with every term in `other`."""
+
+        other_sum = PauliSum.from_terms([(1, other)])
+        return all(
+            PauliSum._commutes(left, right) for left in self._terms for right in other_sum._terms
+        )
+
+    def is_self_commuting(self) -> bool:
```
Comment (speller26) [inline src/braket/quantum_information/pauli_string.py]: Can be a property, and the value determined on instantiation (instead of computing each time). Also, rename to `all_terms_commute`

**Comment by speller26** [comment]
Hi @dorakingx, apologies, at the time of submission, we already had a leading candidate. Thanks for your contribution.

**Comment by dorakingx** [comment]
@speller26 It was a shame this time, but I’m looking forward to the next opportunity.
